components/Collaborate/CollabEditor.js

```diff
@@ -0,0 +1,245 @@
+    fontWeight: 'bold',
+    color: theme.palette.secondary[200],
+  },
+  transcriptFooter: {
+    display: 'flex',
+    flexDirection: 'row',
+    justifyContent: 'flex-end',
+    alignItems: 'center',
+    padding: 0,
+  },
+  transcribeButton: {
+    borderRadius: 30,
+    // border: `1px solid ${theme.palette.primary.main}`,
+  },
+  editButton: {
+    color: theme.palette.secondary[200],
+    borderRadius: 30,
+    // border: `1px solid ${theme.palette.primary.main}`,
+  },
+  newReplyFabIcon: {
+    marginRight: 8,
+    fontSize: '1.2rem',
+  },
+  prosemirrorEditor: {
+    borderRadius: 8,
+    border: `1px solid ${theme.palette.secondary[200]}`,
+    marginTop: 16,
+    marginBottom: 16,
+    '&.hide': { display: 'none' },
+  },
+}));
+
+const colors = [
+  '#ECD444',
+  '#EE6352',
+  '#9E6F21',
+  '#E5F9DB',
+  '#1B9C85',
+  '#068DA9',
+  '#EB455F',
+  '#5D3891',
+  '#BAD1C2',
+];
+
+const color = colors[Math.floor(Math.random() * colors.length)];
+
+const CollabEditor = ({ article }) => {
+  const editor = useRef(null);
+  const [editorView, setEditorView] = useState(null);
+  const currentUser = useCurrentUser();
+  const onTranscribe = () => {
+    if (!currentUser) {
+      return alert(t`Please login first.`);
+    }
+    const ydoc = new Y.Doc();
+    const permanentUserData = new Y.PermanentUserData(ydoc);
+    permanentUserData.setUserMapping(ydoc, ydoc.clientID, currentUser.name);
+    ydoc.gc = false;
+
+    const provider = new HocuspocusProvider({
+      url: PUBLIC_COLLAB_SERVER_URL,
+      name: article.id,
+      broadcast: false,
+      document: ydoc,
+      // onAwarenessChange: ({ states }) => {
+      //   console.log('provider', states);
+      // },
+    });
+    provider.setAwarenessField('user', {
+      name: currentUser.name,
+      color,
+    });
+    const yXmlFragment = ydoc.get('prosemirror', Y.XmlFragment);
+
+    if (editorView) editorView.destroy();
+    setEditorView(
+      new EditorView(editor.current, {
+        state: EditorState.create({
+          schema,
+          doc: DOMParser.fromSchema(schema).parse(editor.current),
+          plugins: [
+            ySyncPlugin(yXmlFragment, { permanentUserData }),
+            yCursorPlugin(provider.awareness),
+            yUndoPlugin(),
+            keymap({
+              'Mod-z': undo,
+              'Mod-y': redo,
+              'Mod-Shift-z': redo,
+            }),
+            PlaceholderPlugin(t`Input transcript`),
+          ].concat(exampleSetup({ schema, menuBar: false })),
+        }),
+      })
+    );
+  };
+
+  const onDone = () => {
+    if (editorView) {
+      let text = '';
+      editorView.state.doc.content.forEach(node => {
+        // console.log(node.textContent);
+        // console.log(node.type.name);
+        if (node.textContent) {
+          text += node.textContent;
+        }
+        text += '\n';
+      });
+
+      // TODO: listen textChanged event?
+      article.text = text;
+      editorView.destroy();
```

Comment:
Question: do we need to do anything to ensure that changes are all synced to Hocuspocus before destroying the prosemirror instance here?